test/index.js

```diff
@@ -21,431 +22,424 @@ const it = lab.it;
 
-        server.register({ register: Reptile, options: { port: port } }, (err) => {
+    const barrier = new Barrier();
+    const server = Hapi.server();
+    const port = await internals.availablePort();
 
-            expect(err).to.not.exist();
+    await server.register({ plugin: Reptile, options: { port } });
 
-            const address = Net.Socket.prototype.address;
-            Net.Socket.prototype.address = function () {
+    const address = Net.Socket.prototype.address;
+    Net.Socket.prototype.address = function () {
 
-                Net.Socket.prototype.address = address;
-                return {
-                    family: 'IPv4',
-                    address: '127.0.0.1'
-                };
-            };
-            const sock = Net.connect(port);
-            let state = 0;
+        Net.Socket.prototype.address = address;
+        return {
+            family: 'IPv4',
+            address: '127.0.0.1'
+        };
+    };
+    const sock = Net.connect(port);
+    let state = 0;
 
-            sock.on('readable', () => {
+    sock.on('readable', () => {
 
-                const buffer = sock.read();
-                if (!buffer) {
-                    return;
-                }
+        const buffer = sock.read();
+        if (!buffer) {
+            return;
+        }
 
-                const result = buffer.toString('ascii');
+        const result = buffer.toString('ascii');
 
-                if (state === 0) {
-                    expect(result.indexOf('>')).to.not.equal(-1);
-                    sock.write('server.hapi\n');
-                }
-                else if (state === 1) {
-                    sock.write('.exit\n');
-                }
-                else if (state === 2) {
-                    done();
-                }
+        if (state === 0) {
+            expect(result.indexOf('>')).to.not.equal(-1);
+            sock.write('server.hapi\n');
+        }
+        else if (state === 1) {
+            sock.write('.exit\n');
+        }
+        else if (state === 2) {
+            barrier.pass();
+        }
 
-                state++;
-            });
-        });
+        state++;
     });
-});
 
-it('creates a REPL that a client can connect to over TCP IPv6', (done) => {
+    return barrier;
+});
 
-    const server = new Hapi.Server();
-    internals.availablePort((port) => {
+it('creates a REPL that a client can connect to over TCP IPv6', async () => {
 
-        server.register({ register: Reptile, options: { port: port } }, (err) => {
+    const barrier = new Barrier();
+    const server = Hapi.server();
+    const port = await internals.availablePort();
 
-            expect(err).to.not.exist();
+    await server.register({ plugin: Reptile, options: { port } });
 
-            const address = Net.Socket.prototype.address;
-            Net.Socket.prototype.address = function () {
+    const address = Net.Socket.prototype.address;
+    Net.Socket.prototype.address = function () {
```

Comment:
Consider making a function to monkey patch this in the future since it is done a few times.